.github/workflows/release-aws-marketplace.yml

```diff
@@ -0,0 +1,105 @@
+# Builds and pushes the proxy docker image to the AWS Marketplace ECR
+#
+# here: https://docs.docker.com/build/ci/github-actions/multi-platform/#distribute-build-across-multiple-runners
+#
+#
+# ECR is a private registry that is not controlled by us
+# Most of the existing Docker actions are not compatible with ECR, so some of this is quite manual
+#
+# Builds for different images are handled using tags, because I could not get other things workings.
+#
+# The tag formats are:
+#   - proxy-{os}-{arch}-{release-tag}
+#   - proxy-{os}-{arch}-build-{timestamp}
+#
+# ECR is immutable - so test builds need a timestamp or they fail with a conflicts
+
+name: "Proxy — Build & Push Proxy Docker Image for AWS MAarketplace"
+
+on:
+  release:
+    types:
+      - published
+  pull_request:
+    branches:
+      - main
+    paths:
+      - .github/workflows/release-aws-marketplace.yml
+
+  workflow_dispatch:
+
+# defaults:
+#   run:
+#     shell: bash -l {0}
+
+env:
+  REGISTRY_IMAGE: cipherstash/cipherstash
+  AWS_REGION: us-east-1
+
+jobs:
+  build:
+    name: 🏗️ Build binaries + Docker images
+    permissions:
+        contents: read
+        packages: write
+        id-token: write # This is required for requesting the JWT
+    strategy:
+      fail-fast: false
+      matrix:
+        build:
+          - { os: linux-arm64-public, arch: linux/arm64, tag: linux-arm64, cache-provider: github }
+    runs-on: ${{matrix.build.os}}
+    steps:
+
+      - name: install-aws-cli
+        uses: unfor19/install-aws-cli-action@v1
+        if: ${{ matrix.build.arch == 'linux/arm64' }}
+        with:
+            version: 2                         # default
+            verbose: false                     # default
+            arch: arm64                        # allowed values: amd64, arm64
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ vars.AWS_MARKETPLACE_IAM_ROLE }}
+          aws-region: us-east-1
+
+      - name: Log in to AWS Marketplace ECR
+        id: ecr-login
+        uses: aws-actions/amazon-ecr-login@v2
+        with:
+          registries: ${{ vars.AWS_MARKETPLACE_ECR_ID }}
+
+      - uses: actions/checkout@v4
+
+      - name: Setup Rust cache
+        uses: Swatinem/rust-cache@v2
+        with:
+          cache-provider: ${{matrix.build.cache-provider}}
+          cache-all-crates: true
+
+      - uses: jdx/mise-action@v2
+        with:
+          version: 2024.12.0 # [default: latest] mise version to install
```

Comment:
Note that this is a different mise version (`2024.12.0`) to what we're using elsewhere (`2025.1.6`). 

We should upgrade this workflow to `2025.1.6` before we merge.

Reply:
Good catch. I pulled it from current `release.yml`, will update there too